databroker/core.py

```diff
@@ -1312,10 +1317,9 @@ def _load(self):
         # still in progress. If it returns None, pass that through.
         if self._run_stop_doc is None:
             stop = self._get_run_stop()
-            if stop is None:
-                self._run_stop_doc = stop
```

Comment:
If we are here, we have already checked that `stop is None` and `self._run_stop_doc is None`, so setting it here has no effect.